aws/resource_aws_efs_file_system_test.go

```diff
@@ -261,6 +283,51 @@ func TestAccAWSEFSFileSystem_ThroughputMode(t *testing.T) {
 	})
 }
 
+func TestAccAWSEFSFileSystem_lifecyclePolicy(t *testing.T) {
+	resourceName := "aws_efs_file_system.foo-with-lifecycle-policy"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckEfsFileSystemDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSEFSFileSystemConfigWithLifecyclePolicy(
+					"transition_to_ia",
+					"invalid_value",
+				),
+				ExpectError: regexp.MustCompile(`got invalid_value`),
+			},
+			{
+				Config: testAccAWSEFSFileSystemConfigWithLifecyclePolicy(
+					"transition_to_ia",
+					efs.TransitionToIARulesAfter30Days,
+				),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckEfsFileSystem(resourceName),
+					testAccCheckEfsFileSystemLifecyclePolicy(resourceName, "badExpectation"),
+				),
+				ExpectError: regexp.MustCompile(`Expected: badExpectation`),
```

Comment:
This test step looks to be from debugging and should be removed 😄

Reply:
Actually that was intended. I wanted to test the error checking code and get better test coverage.  Is that no bueno?